src/VisualStudio/CSharp/Test/CodeModel/AbstractFileCodeElementTests.cs

```diff
@@ -14,16 +14,30 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.UnitTests.CodeModel
     /// </summary>
     public abstract class AbstractFileCodeElementTests : IDisposable
     {
-        private readonly Tuple<TestWorkspace, FileCodeModel> _workspaceAndCodeModel;
+        private readonly string _contents;
+        private Tuple<TestWorkspace, FileCodeModel> _workspaceAndCodeModel;
+
+        public AbstractFileCodeElementTests(string contents)
+        {
+            _contents = contents;
+        }
+
+        public Tuple<TestWorkspace, FileCodeModel> WorkspaceAndCodeModel
+        {
+            get
+            {
+                return _workspaceAndCodeModel ?? (_workspaceAndCodeModel = CreateWorkspaceAndFileCodeModelAsync(_contents));
```

Comment:
Constructors are called before `UseExportProviderAttribute`, so eager initialization of these properties will throw an exception on the following line (the entire gap from the end of a test to the initialization of the export provider for the next test is attributed to the former test in the exception message):

https://github.com/dotnet/roslyn/blob/7eb0b2dd998713278e9f34110492901b3ac45eeb/src/Workspaces/CoreTestUtilities/UseExportProviderAttribute.cs#L74

Reply:
> Constructors are called before UseExportProviderAttribute

Interesting. That feels like an xunit bug/gotcha -- is that something we can fix or at least make more diagnosable? Otherwise the obvious sequence of events of:

1. Write a new test from scratch, get error that you don't have attribute.
2. Apply attribute.
3. Run test.
4. Still broken.

seems quite undebuggable.

Reply:
➡️ Extracted a method which throws the `InvalidOperationException`, and added a comment to help users who hit it.